biotrainer/embedders/embedding_service.py

```diff
@@ -230,6 +232,42 @@ def _save_and_reset_embeddings(self, embeddings: Dict[str, ndarray], last_save_i
         del embeddings
         return last_save_id, {}
 
+    @staticmethod
+    def reduce_embeddings_dimension(
```

Comment:
The name might be confusing because per-protein embeddings are also referred to as "reduced". Maybe call it "def embeddings_dimensionality_reduction"?